### PR TITLE
feat: Add config setting to control how many List items are printed

### PR DESCRIPTION
**Pull request by Walnut356**
I also modified the test to make sure it works properly. For the python function/docstring, I tried my best to copy the style/functionality in the functions around it, but please let me know if I made any mistakes ^^

### CODE REVIEW

**Review by stinodego — CHANGES_REQUESTED** [review]
I would be happy to have this option added - I have looked for it before and it didn't exist.

I do think we need a different name though. Maybe `FMT_TABLE_CELL_LIST_LEN`?

**Comment by avimallu** [comment]
Will this also be available as a config?

**Comment by stinodego** [comment]
> Will this also be available as a config?

Yes!

**Review by stinodego — APPROVED** [review]
Thanks for the addition! Maybe someone else can judge the Rust implementation.

File: crates/polars-core/src/fmt.rs [hunk 1]
```diff
@@ -949,12 +949,32 @@ impl Series {
...
+                for (i, item) in self.iter().enumerate() {
+                    result.push_str(&item.to_string());
+
+                    if i != self.len() - 1 {
+                        result.push_str(", ");
+                    }
+
+                    if i == max_items - 2 {
+                        result.push_str("… ");
+                        result.push_str(&self.get(self.len() - 1).unwrap().to_string());
```
Comment (ritchie46) [inline crates/polars-core/src/fmt.rs:970]: I think we should use a `write!` macro here, to elide the superfluous `to_string` call.

**Review by ritchie46 — CHANGES_REQUESTED** [review]

File: crates/polars-core/src/fmt.rs (hunk 1 above)
Comment (Walnut356) [reply crates/polars-core/src/fmt.rs]: for sure :smile:

**Comment by ritchie46** [comment]
Thank you @Walnut356

**Comment by cmdlineluser** [comment]
Thank you for this, not being able to control this was somewhat annoying.

I just noticed a couple of things which may not be intended, with `-1` the `...` placeholder is still added.

`0` and `1` seem to print all items:
```python
import polars as pl

df = pl.DataFrame({"nums": [[1, 2, 3, 4, 5, 6]]})

with pl.Config(fmt_table_cell_list_len=-1): df
# shape: (1, 1)
# ┌──────────────────────┐
# │ nums                 │
# │ ---                  │
# │ list[i64]            │
# ╞══════════════════════╡
# │ [1, 2, 3, 4, 5, … 6] │
# └──────────────────────┘

with pl.Config(fmt_table_cell_list_len=0): df
# shape: (1, 1)
# ┌────────────────────┐
# │ nums               │
# │ ---                │
# │ list[i64]          │
# ╞════════════════════╡
# │ [1, 2, 3, 4, 5, 6] │
# └────────────────────┘

with pl.Config(fmt_table_cell_list_len=1): df
# shape: (1, 1)
# ┌────────────────────┐
# │ nums               │
# │ ---                │
# │ list[i64]          │
# ╞════════════════════╡
# │ [1, 2, 3, 4, 5, 6] │
# └────────────────────┘
```

**Pull request merged**